dashboard/modules/metrics/metrics_head.py

```diff
@@ -266,9 +277,26 @@ def _create_default_prometheus_configs(self):
         os.makedirs(os.path.dirname(prometheus_config_output_path), exist_ok=True)
         shutil.copy(PROMETHEUS_CONFIG_INPUT_PATH, prometheus_config_output_path)
 
+    @dashboard_utils.async_loop_forever(METRICS_RECORD_INTERVAL_S)
+    async def record_dashboard_metrics(self):
+        dashboard_proc = psutil.Process()
+        self._dashboard_head.metrics.metrics_dashboard_cpu.labels(
+            ip=self._ip,
+            pid=self._pid,
+            Component=self._component,
+            SessionName=self._session_name,
+        ).set(float(dashboard_proc.cpu_percent()) * 100)
+        self._dashboard_head.metrics.metrics_dashboard_mem.labels(
+            ip=self._ip,
+            pid=self._pid,
+            Component=self._component,
+            SessionName=self._session_name,
+        ).set(float(dashboard_proc.memory_info().rss) / 1.0e6)
```

Comment:
seems like memory_monitor is using (rss - shared), perhaps we should just use uss here, and oom killer will do the same?

Reply:
Yeah I think using uss makes sense. Another thing we can try is to compare USS and RSS - shared and see how much they are different from some nightly tests...